src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -275,6 +276,14 @@ private static MsalServiceException ExtractErrorsFromTheResponse(HttpResponse re
                 return null;
             }
 
+            if (response.HeadersAsDictionary != null && response.HeadersAsDictionary.ContainsKey(ThrottleCommon.ThrottleRetryAfterHeaderResponseValue))
```

Comment:
why does the code need to be duplicated?

Reply:
it does not, this is a draft and I am still in the process of refactoring. will be removed